lib/handlers/pools/pool-caching/v2/v2-dynamo-cache.ts

```diff
@@ -18,6 +18,19 @@ export class V2DynamoCache implements ICache<{ pair: Pair; block?: number }> {
       },
     })
   }
+
+  async batchGet(keys: Set<string>): Promise<Record<string, { pair: Pair; block?: number | undefined } | undefined>> {
+    const valuesPromise = Array.from(keys).map((key) => this.get(key).then((value) => ({ key, value })))
+    const keyValuePairs = await Promise.all(valuesPromise)
+    const records: Record<string, { pair: Pair; block?: number | undefined } | undefined> = {}
+
+    keyValuePairs.forEach(({ key, value }) => {
+      records[key] = value
+    })
+
+    return records
+  }
+
```

Comment:
Sure we can. I'm thinking for the aws `batchGet` implementation, we don't implement out the handling of `unprocessedKey` to begin with, since the input token pair arrays to fetch pools caching shouldn't exceed 16MB. wdyt?

Reply:
And right now sor doesn't use `batchGet` yet.

Reply:
yeah... my main concern with firing multiple network calls at the same time is that it could be a bit slower, and in this case it might make sense to do batchGet since we are fetching from the same DB.

Yeah, we could add metrics to unprocessedKey just to keep track of it (and maybe set some alarm in case it ever starts showing up).

Reply:
Yep, I can see the latencies concern, since we've been bite by router so bad.. Right now I think the prod has the same latencies as it could be from this PR, since it's parallelizing dynamo `get` instead of calling `batchGet` under v2 `pool-provider`. Therefore I think it makes sense to created linear ticket to follow up.

Reply:
at the very least a TODO with a linear ticket attached

Reply:
With the additional time for dev, implemented the AWS `batchGet` here. Unfortunately, we still need a TODO with ticket, because `ICache.batchGet(keys: Set<string>)` only allows passing in simple key, meanwhile we need composite key for partition key & sort key. I created https://linear.app/uniswap/issue/ROUTE-83 to follow up.